asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,412 @@
+        signer: await newSigner(keyDirectoryPathOrg1),
+    });
+
+    // The gRPC client connection from org2 should be shared by all Gateway connections to this endpoint.
+    const clientOrg2 = await newGrpcConnection(
+        tlsCertPathOrg2,
+        peerEndpointOrg2,
+        peerContainerOrg2
+    );
+
+    const gatewayOrg2 = connect({
+        client: clientOrg2,
+        identity: await newIdentity(certPathOrg2, mspIdOrg2),
+        signer: await newSigner(keyDirectoryPathOrg2),
+    });
+
+    try {
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg1 = gatewayOrg1.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg1 = networkOrg1.getContract(chaincodeName);
+
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg2 = gatewayOrg2.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg2 = networkOrg2.getContract(chaincodeName);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Create new assets on the ledger.
+        await createAsset(contractOrg1);
+
+        //Read  asset from the org1's private data collection.
+        await readAssetPrivateDetails(contractOrg1,org1PrivateCollectionName,assetID1);
+
+        //Attempt to transfer asset without prior aprroval from org2, transaction expected to fail.
+        await transferAssetWithoutAprroval(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Read the asset by ID.
+        await readAssetByID(contractOrg2);
+
+        //Make agreement to transfer the asset from org1 to org2.
+        await agreeToTransfer(contractOrg2);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        //Read transfer agreement.
+        await readTransferAgreement(contractOrg1);
+
+        // Tranfer asset to org2.
+        await transferAsset(contractOrg1);
+
+        //Again ReadAsset : results will show that the buyer identity now owns the asset.
+        await readAssetByID(contractOrg1);
+
+        //Confirm that transfer removed the private details from the Org1 collection.
+        await checkPrivateDataExist(contractOrg1);
+
+        // Delete AssetID2 as Org1.
+        await deleteAsset(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Org2 can ReadAssetPrivateDetails: Org2 is owner, and private details exist in new owner's Collection
+        await readAssetPrivateDetails(contractOrg2,org2PrivateCollectionName,assetID1);
+
+    } finally {
+        gatewayOrg1.close();
+        clientOrg1.close();
+
+        gatewayOrg2.close();
+        clientOrg2.close();
+    }
+}
+
+main().catch((error) =>
+    console.error('******** FAILED to run the application:', error)
```

Comment:
If you care about the process exit code, you can set it here as [process.exitCode](https://nodejs.org/api/process.html#processexitcode_1) and then let the process reach the end of execution gracefully